tracing-subscriber/src/registry/sharded.rs

```diff
@@ -89,6 +89,24 @@ fn id_to_idx(id: &Id) -> usize {
     id.into_u64() as usize - 1
 }
 
+pub(crate) struct LayerGuard<'a> {
+    id: Id,
+    registry: &'a Registry,
+}
+
+impl<'a> Drop for LayerGuard<'a> {
+    fn drop(&mut self) {
+        CLOSE_COUNT.with(|count| {
+            let c = count.get();
+            if c > 0 {
+                count.set(c - 1);
+            } else {
+                self.registry.spans.remove(id_to_idx(&self.id));
+            }
+        })
+    }
+}
```

Comment:
super nitpicky, but i might move this lower in the file, after the `Registry` impls?

Reply:
Moved next to DataInner's drop impl.